WooCommerce/Classes/ViewRelated/Coupons/CouponListViewController.swift

```diff
@@ -179,6 +197,26 @@ private extension CouponListViewController {
         footerSpinnerView.stopAnimating()
         tableView?.tableFooterView = footerEmptyView
     }
+
+    /// Triggers the coupon creation flow
+    ///
+    func startCouponCreation(discountType: Coupon.DiscountType) {
+        let viewModel = viewModel.createAddEditCouponViewModel(with: discountType) { createdCoupon, shareMessage in
+            self.showCouponCreationSuccess(couponCode: createdCoupon.code, shareMessage: shareMessage)
+        }
```

Comment:
Also - I think we can simplify this part just so we don't have to keep a reference of `addEditHostingController` in this controller. Here's my suggestion:
- In `AddEditCouponViewModel`, when the creation is successful, we can ask the `AddEditCoupon` view to present the success view.
- When the success view is dismissed, it triggers a callback to let the `AddEditCoupon` know that it's time to get back to the list. The view will then trigger the view model's `onCompletion` closure, which asks the `CouponListViewController` to dismiss the hosting controller and reload the list. 

I see that this solution will probably cause 2 dismiss animations though 🤔 WDYT?

Reply:
Following your suggestion, I actually managed to change this call stack in a way that doesn't trigger two dismiss animations, unless we use the swipe down gesture to dismiss the coupon creation success view since it will depend on the `onDisapper` SwiftUI event. Let me know what you think!

Reply:
I missed this conversation yesterday 😅 Your solution looks good! 💯